flow/templates/anvil.sh

```diff
@@ -0,0 +1,33 @@
+{# Templated in accordance with: https://www.rcac.purdue.edu/knowledge/anvil/ #}
+{% extends "slurm.sh" %}
+{% block tasks %}
+    {% if resources.ngpu_tasks %}
+        {% if not ('gpu' in partition or force) %}
+            {% raise "GPU operations require a gpu partition!" %}
+        {% endif %}
+        {% if resources.ngpu_tasks > 4 %}
+            {% raise "Cannot request a gpu job with more than 4 GPUs." %}
+        {% endif %}
+    {% else %}
+        {% if 'gpu' in partition and not force %}
+            {% raise "Requesting gpu partition, but no GPUs requested!" %}
+        {% endif %}
+    {% endif %}
+    {% if partition in ['debug', 'shared', 'highmem'] and resources.ncpu_tasks > 128 %}
+        {% raise "Cannot request a shared partition with more than 128 CPU tasks." %}
+    {% endif %}
```

Comment:
This should be unnecessary given `env._shared_partitions`. The reason I didn't remove it from other templates is either some clusters only allow shared submissions of one socket not the entire node and I may have forgotten :sweat_smile: , but in this case this is more than one node so you should be good.
```suggestion
```

Reply:
I don't know what you mean about `env._shared_partitions`. If I remove these lines, flow generates a 2 node shared partition script without error or warning:
```
#SBATCH --partition=shared
#SBATCH -N 2
#SBATCH --ntasks=256
#SBATCH -A dmr140129
```
Is your intention that the user receive this particular error from `sbatch` instead?

Reply:
I removed the checks so that `sbatch` provides the user with errors on invalid resource requests for a given partition.